Classes/Mapper/Event.php

```diff
@@ -174,15 +221,43 @@ public function findForAutomaticStatusChange(): Collection
      */
     public function findForRegistrationDigestEmail(): Collection
     {
-        $whereClause = 'registrations <> 0' .
-            ' AND object_type <> ' . \Tx_Seminars_Model_Event::TYPE_TOPIC .
+        $whereClause = 'registrations <> ? AND object_type <> ? AND hidden = ? AND deleted = ? ' .
             ' AND EXISTS (' .
             'SELECT * FROM tx_seminars_attendances ' .
-            'WHERE tx_seminars_attendances.deleted = 0 ' .
+            'WHERE tx_seminars_attendances.deleted = ? ' .
             ' AND tx_seminars_attendances.seminar = tx_seminars_seminars.uid' .
             ' AND tx_seminars_attendances.crdate > tx_seminars_seminars.date_of_last_registration_digest' .
             ')';
 
-        return $this->findByWhereClause($whereClause, 'begin_date ASC');
+        $sql = 'SELECT * FROM ' . $this->getTableName() . ' WHERE ' . $whereClause . ' ORDER BY begin_date ASC';
+
+        /** @var Connection $connection */
+        $connection = $this->getConnectionForTable($this->getTableName());
+        $statement = $connection->prepare($sql);
+        $statement->bindValue(1, 0, \PDO::PARAM_INT);
+        $statement->bindValue(2, \Tx_Seminars_Model_Event::TYPE_TOPIC, \PDO::PARAM_INT);
+        $statement->bindValue(3, 0, \PDO::PARAM_INT);
+        $statement->bindValue(4, 0, \PDO::PARAM_INT);
+        $statement->bindValue(5, 0, \PDO::PARAM_INT);
+        $statement->execute();
+        $rows = $statement->fetchAll();
+
+        return $this->getListOfModels($rows);
+    }
+
+    private function getOrderByFromTca(): array
```

Comment:
I find this very hard to read: one or two return array elements (either 0, or 1, or both).

Instead, I propose we rebuild this to
```php
private function addSortingForBackend(QueryBuilder $query): void;
```

(with the `: void` return type declaration being omitted in the real code for PHP 7.0 compatibilty)